common/src/main/java/org/jboss/hal/testsuite/cli/DomainManager.java

```diff
@@ -472,4 +475,105 @@ public boolean waitUntilAvailable(String host, long timeout) {
         }
         return isRunning;
     }
+
+    @Override
+    public void reloadIfRequiredAndWaitUntilRunning(long timeout) {
+        if (isReloadRequired(listRunningServers(getDefaultHost()))) {
+            reloadAndWaitUntilRunning(timeout);
+        }
+    }
+
+    @Override
+    public void reloadAndWaitUntilRunning(long timeout) {
+        List<String> servers = filterAllRunningServers(listRunningServers(getDefaultHost()));
```

Comment:
You are right, it is just confusing name. I would leave the refactoring to different PR if decided to be done.